js/src/common/components/Dropdown.js

```diff
@@ -53,13 +54,25 @@ export default class Dropdown extends Component {
     // bottom of the viewport. If it does, we will apply class to make it show
     // above the toggle button instead of below it.
     this.$().on('shown.bs.dropdown', () => {
+      const { lazyDraw, onshow } = this.attrs;
+
       this.showing = true;
 
-      if (this.attrs.onshow) {
-        this.attrs.onshow();
+      // If using lazy drawing, redraw before calling `onshow` function
+      // to make sure the menu DOM exists in case the callback tries to use it.
+      if (lazyDraw) {
+        m.redraw.sync();
```

Comment:
Could we get around this complexity by attaching the onshow to the dropdown content vnode as an oncreate attr?

Reply:
Are you suggesting keeping the `if (!lazyDraw)` bellow, and simply move this one to the oncreate? 

The oncreate would then need to be conditional for `lazyDraw`, else it would be called on Dropdown render for non-lazy drawn dropdowns.

Am I making sense 🤔

Reply:
Not quite. What if we wrap `this.getMenu(items)` in a fragment or div/span, and attached the body of this callback to that fragment/div/span as an `oncreate`? Then, the onshow and redraw logic would only run whenever the element is opened, without relying on bootstrap, or having to account separately for lazy vs non-lazy (which is a potential point of failure)

Reply:
Unless I'm misunderstanding, wouldn't that change the behavior? With non-lazy, the component *will* be rendered, therefore the `oncreate` would run when the entire Dropdown is created.

Reply:
We're still technically changing behavior of a public component 🤷‍♂️

And we haven't defined public/private API anywhere for developers to know what might break and what shouldn't...